src/pylibsshext/scp.pyx

```diff
@@ -122,7 +122,9 @@ cdef class SCP:
             size = libssh.ssh_scp_request_get_size(scp)
             mode = libssh.ssh_scp_request_get_permissions(scp)
 
-            read_buffer = <char *>PyMem_Malloc(size)
+            # cap the buffer size to reasonable number -- libssh will not return the whole data at once anyway
+            read_buffer_size = min(size, 65536)
```

Comment:
Let's name this piece of data (though, I'd probably even move it into a module-level constant):
```suggestion
            default_read_buffer_size = 65536
            read_buffer_size = min(size, default_read_buffer_size)
```

Is there a constant within libssh that we could reuse instead of re-hardcoding it here?

Reply:
Is it possible to rely on this https://github.com/canonical/libssh/blob/6f4ec9be/src/connector.c#L33?

Reply:
I think we will have eventually the same problem with writing (but that is likely much less used in the context of ansible) so I will make a separate constant for this .